hack/e2e.sh

```diff
@@ -2,11 +2,9 @@
 
 set -euo pipefail
 
-GOPATH="$(mktemp -d)"
-export GOPATH
+tmp="$(mktemp -d)"
+trap 'rm -rf "$tmp"' EXIT
```

Comment:
I'd personally prefer we drop this. This is almost exclusively used in CI where it doesn't matter if we clean this up, and we might want it to stick around so we can see what we built.